snapcraft/plugins/jhbuild.py

```diff
@@ -0,0 +1,1035 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The JHBuild plugin is used for GNOME-based parts.
+
+JHBuild is a tool used to build components of the GNOME ecosystem from version
+control. See the JHBuild manual for more details:
+https://developer.gnome.org/jhbuild/stable/.
+
+The plugin can be customized with the following keys:
+
+  - modules (required): a list of modules to include in the part, modules that
+                        should be skipped should be prefixed with '-'
+  - moduleset: the module set JHBuild should use (default: gnome-world)
+  - snap-name (required): the name of the snap this part is included in
```

Comment:
This will always be $SNAPCRAFT_PROJECT_NAME. We probably can drop this attribute.